### PR TITLE
Fix(eos_validate_state): Fix the DHCP not recognized error for STUN and Connectivity tests

### LINKED ISSUE #4725: Bug(eos_validate_state): DHCP not recognized for Pathfinder interfaces
### Issue Summary

In a Pathfinder configuration, if an Edge node has the interface set for DHCP, eos_validate_state gives an error for that node and cannot create any reports. 

### Which component(s) of AVD impacted

eos_validate_state

### How do you run AVD ?

Ansible CLI (with virtual-env or native python)

### Steps to reproduce

Create a Pathfinder network with an Edge device that has an interface set for DHCP: 

```yaml


    - name: Edge22
      id: 22
      cv_pathfinder_region: REGION2
      cv_pathfinder_site: SITE22
      l3_interfaces:
        - name: Ethernet2
          ip_address: 172.16.22.2/24
          wan_carrier: mpls_r2
          flow_tracking:
            enabled: true
        - name: Ethernet3
          ip_address: dhcp
          wan_carrier: internet
          dhcp_accept_default_route: true
          peer_ip: 22.22.22.1
          dhcp_ip: 22.22.22.10
          flow_tracking:
            enabled: true
          cv_pathfinder_internet_exit: 
            policies:
              - name: DEFAULT_INTERNET_EXIT
```

Then run a playbook to validate state. This is the playbook I used: 

```yaml
---
- name: Validate state for Pathfinder
  hosts: PF_NETWORK
  tasks:
    - name: Validate states on EOS devices
      ansible.builtin.import_role:
        name: arista.avd.eos_validate_state
```

You can find a full topology here: https://github.com/tonybourkesdnpros/V3_Repo_Pathfinder.git

In the Pathfinder_MR directory, you'll find four edges configured via DHCP in Region2. 

It doesn't appear that validate_state recognizes DHCP as a valid option. DHCP will probably be fairly common in Pathfinder deployments.
```


### Relevant log output

```shell
TASK [arista.avd.eos_validate_state : Run eos_validate_state_runner] *********************************************************************************************************************************************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ValueError: 'dhcp' does not appear to be an IPv4 or IPv6 interface
fatal: [Edge20]: FAILED! => {"changed": false, "msg": "Error during plugin execution: 'dhcp' does not appear to be an IPv4 or IPv6 interface"}
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ValueError: 'dhcp' does not appear to be an IPv4 or IPv6 interface
fatal: [Edge22]: FAILED! => {"changed": false, "msg": "Error during plugin execution: 'dhcp' does not appear to be an IPv4 or IPv6 interface"}
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ValueError: 'dhcp' does not appear to be an IPv4 or IPv6 interface
fatal: [Edge23]: FAILED! => {"changed": false, "msg": "Error during plugin execution: 'dhcp' does not appear to be an IPv4 or IPv6 interface"}
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ValueError: 'dhcp' does not appear to be an IPv4 or IPv6 interface
fatal: [Edge24]: FAILED! => {"changed": false, "msg": "Error during plugin execution: 'dhcp' does not appear to be an IPv4 or IPv6 interface"}
```

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by MaheshGSLAB**
## Change Summary

Fix the DHCP not recognized error for STUN and Connectivity tests. If the interface is dhcp then AvdTestStun and AvdTestP2PIPReachability tests try to collect the ip address from interface cause below error.
```
TASK [arista.avd.eos_validate_state : Run eos_validate_state_runner] *********************************************************************************************************************************************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ValueError: 'dhcp' does not appear to be an IPv4 or IPv6 interface
fatal: [Edge20]: FAILED! => {"changed": false, "msg": "Error during plugin execution: 'dhcp' does not appear to be an IPv4 or IPv6 interface"}
```

## Related Issue(s)

Fixes #4725 

## Component(s) name

`arista.avd.eos_validate_state`

## Proposed changes
Fix the DHCP not recognized error for STUN and Connectivity tests. If the interface is dhcp then AvdTestStun and AvdTestP2PIPReachability tests try to collect the ip address from interface cause below error.

Added the logic to check if interface is dhcp interface or not to avoid error.

## How to test
Make the interface as DHCP and run the molecule. It should skip the test with below warning:
```
Host 'host1' interface 'Ethernet2' IP address is unavailable. AvdTestP2PIPReachability is skipped.
Host 'host1' interface 'Ethernet4' IP address is unavailable. AvdTestStun is skipped.
```

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/molecule/eos_validate_state/intended/test_catalogs/dc2-leaf3a.arista.com-catalog.yml [hunk 1]
```diff
@@ -73,6 +73,13 @@ anta.tests.connectivity:
       port: Ethernet15
     result_overwrite:
       custom_field: 'Local: Ethernet15 - Remote: dc2-leaf2b Ethernet2'
+- VerifyLLDPNeighbors:
+    neighbors:
+    - neighbor_device: dc2-leaf3b.arista.com
+      neighbor_port: Ethernet16
+      port: Ethernet16
+    result_overwrite:
+      custom_field: 'Local: Ethernet16 - Remote: dc2-leaf3b.arista.com Ethernet16'
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_validate_state/intended/test_catalogs/dc2-leaf3a.arista.com-catalog.yml:82]: hehe given that this is a remote interface we control and it is shut down on leaf3b this test will always fail.. not sure if we should think (not in this PR ;) ) to tackle this later (means need access to our peer structured_config) to detect this. @carl-baillargeon for info

**Review by gmuloc — APPROVED** [review]

**Comment by gmuloc** [comment]
@tonybourkesdnpros if you have time to test the PR it could be great!

File: ansible_collections/arista/avd/molecule/eos_validate_state/intended/test_catalogs/dc2-leaf3a.arista.com-catalog.yml (hunk 1 above)
Comment (carl-baillargeon) [reply ansible_collections/arista/avd/molecule/eos_validate_state/intended/test_catalogs/dc2-leaf3a.arista.com-catalog.yml:82]: @gmuloc Yes good point. I will open an issue to track this.

**Review by carl-baillargeon — APPROVED** [review]
LGTM

**Pull request merged**